### PR TITLE
calendarspec: allow repetition values with ranges

### PR DESCRIPTION
**Pull request by dobyrch**
This is my last addition to the calendar syntax, I promise.

Allowing a step with ranges is the last bit of syntax needed to bring
systemd timers up to par with Vixie cron.  In addition to easing the
translation of crontabs to `OnCalendar=` specifications, generalizing
ranges to work with a step actually simplifies the code in a few places.

The semantics of ranges (hopefully) haven't changed, although I
introduced an additional restriction for dates scheduled relative to the
end of the month: `*-*~29`, `*-*~30`, and `*-*~31` are no longer valid.
Handling every edge case added more complexity than warranted by such
a dubious use case (it's not often people want to schedule something on
"the 30th last day of the month").

The second commit is purely aesthetic; it renames `value`/`range_end` to
`start`/`stop` since the name `value` seemed rather ambiguous after these
additions.

### CODE REVIEW

**Comment by poettering** [comment]
> This is my last addition to the calendar syntax, I promise.

That's a pity! I kinda liked the love you gave the calendaring syntax!

File: src/basic/calendarspec.c [hunk 1]
```diff
@@ -157,20 +175,32 @@ _pure_ static bool chain_valid(CalendarComponent *c, int from, int to, bool end_
         if (!c)
                 return true;
 
+        /* Forbid dates more than 28 days from the end of the month */
+        if (end_of_month)
+                to -= 3;
```
Comment (poettering) [inline src/basic/calendarspec.c:178]: it might make sense to document this in the man page, dunno

File: src/basic/calendarspec.c [hunk 2]
```diff
@@ -1014,11 +1019,24 @@ int calendar_spec_from_string(const char *p, CalendarSpec **spec) {
         return r;
 }
 
+static int find_end_of_month(struct tm *tm, bool utc, int day)
+{
```
Comment (poettering) [inline src/basic/calendarspec.c:1020]: nitpick: we place the opening bracket on the same line as the function name...

**Comment by poettering** [comment]
merging, my two nitpicks are minor.

**Comment by dobyrch** [comment]
>> is my last addition to the calendar syntax, I promise.

> That's a pity! I kinda liked the love you gave the calendaring syntax!

Well, "last addition" until the next RFE or bug shows up, and then I'll pop in again :)

**Pull request merged**